src/Features/Core/Portable/NavigateTo/NavigateToSearcher.cs

```diff
@@ -25,9 +25,7 @@ internal class NavigateToSearcher
         private readonly IAsynchronousOperationListener _asyncListener;
         private readonly INavigateToSearchCallback _callback;
         private readonly string _searchPattern;
-        private readonly bool _searchCurrentDocument;
         private readonly IImmutableSet<string> _kinds;
-        private readonly Document? _currentDocument;
```

Comment:
we used to set this state, and then try to do the normal full search, bailing out if we saw this.  it was much easier to just not have this state and just directly handle teh 'search current doc' codepath right up front.